app/controllers/escorts_controller.rb

```diff
@@ -16,14 +16,28 @@ def create
   def show
   end
 
+  def confirm_cancel
+    @form = Forms::Escort.new(escort)
+  end
+
+  def cancel
+    @form = Forms::Escort.new(escort)
+    if @form.validate(escort_params)
+      escort.cancel!(current_user, escort_params[:cancelling_reason]) unless escort.issued? || escort.cancelled?
+      redirect_to root_path
```

Comment:
As discussed offline, when the escort can no longer be cancelled, the user is just redirected to the `root_path` with no information whatsoever of what happened (was the action successful or unsuccessful?). Relying on having a user interface that hides (visually only) the ability to access this action seems bad practice 😰, the action should do whatever needs to do regardless of the user interface associated with it.